java-checks/src/main/java/org/sonar/java/checks/IgnoredTestsCheck.java

```diff
@@ -43,5 +60,22 @@ public void visitNode(Tree tree) {
     if (ignoreAnnotationValues != null && ignoreAnnotationValues.isEmpty()) {
       reportIssue(methodTree.simpleName(), "Fix or remove this skipped unit test");
     }
+    BlockTree block = methodTree.block();
+    if(block != null) {
```

Comment:
Here we only visit Tree.Kind.METHOD and not Tree.Kind.CONSTRUCTOR, so we have this false-negative:
```
class MyTest {
  MyTest() {
    Assume.assumeTrue(false); // false-negative
  }
  // ... tests
}
```

Reply:
Wow... really a corner case IMO but indeed. Case covered.